lib/segment/src/types.rs

```diff
@@ -771,6 +771,9 @@ pub struct VectorDataConfig {
     pub index: Indexes,
     /// Vector specific quantization config that overrides collection config
     pub quantization_config: Option<QuantizationConfig>,
+    /// Enable multiple vectors per point
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub multivec: Option<bool>,
```

Comment:
I want to discuss `bool` type here. I know that this option is documented in the design document. But we can discuss the way to use some `enum` instead of bool for future extensions.

Reply:
What kind of extensions do you have in mind?

Reply:
For instance,
```
pub struct VectorDataConfig {
    ...
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub multivec: Option<MultivectorConfig>,
}

pub enum MultivectorConfig {
    #[default]
    Colbert(), // or some another placeholder name
}
```
It allows us to extend multivector parameters if needed.
Sorry that I propose it after opening this PR, I just realized it right now.

Reply:
Thanks for the suggestion, I have added an enum to configure the similarity for multivectors :+1: